packages/builder/src/components/backend/DatasourceNavigator/modals/CreateDatasourceModal.svelte

```diff
@@ -99,6 +100,28 @@
     }
     integrations = newIntegrations
   }
+
+  function sortIntegrations(integrations) {
+    let integrationsArray = Object.entries(integrations)
+    integrationsArray.sort((a, b) => {
```

Comment:
NAB: Maybe just a little bit easier to read if you break the inner function here one level above:
```js
function sortIntegrations(integrations) {
    let integrationsArray = Object.entries(integrations)
    function getTypeOrder(schema) {
       if (schema.type === DatasourceTypes.API) {
         return 1
       }
       if (schema.type === DatasourceTypes.RELATIONAL) {
         return 2
       }
      return schema.type?.charCodeAt(0)
    }
       
    integrationsArray.sort((a, b) => {
      let typeOrderA = getTypeOrder(a[1])
      let typeOrderB = getTypeOrder(b[1])
      if (typeOrderA === typeOrderB) {
        return a[1].friendlyName?.localeCompare(b[1].friendlyName)
      }
      return typeOrderA < typeOrderB ? -1 : 1
    })
    return integrationsArray
  }
  ```